packages/mobile/src/harmony-native/components/input/SelectablePill/SelectablePill.tsx

```diff
@@ -1,137 +1,164 @@
+  const {
+    icon: Icon,
+    label,
+    size = 'small',
+    isSelected: isSelectedProp,
+    disabled,
+    onPress,
+    ...other
+  } = props
   const { color } = useTheme()
+  const pressed = useSharedValue(0)
+  const selected = useSharedValue(0)
+  const [isPressing, setIsPressing] = useState(false)
+  const [isSelected, setIsSelected] = useState(isSelectedProp)
+
+  const handlePressIn = useCallback(() => {
+    setIsPressing(true)
+  }, [])
+
+  const handlePress = useCallback(
+    (e: GestureResponderEvent) => {
+      onPress?.(e)
+      setIsPressing(false)
+      setIsSelected((isSelected) => !isSelected)
+    },
+    [onPress]
+  )
 
-  const selectValue = useDerivedValue(() => {
-    return withTiming(isPressing || props.isSelected ? 1 : 0, {
-      duration: 120,
-      easing: Easing.bezier(0.44, 0, 0.56, 1)
+  const tap = Gesture.Tap()
+    .onBegin(() => {
+      pressed.value = withTiming(1, animationConfig)
+      if (!isSelected) {
+        selected.value = withTiming(1, animationConfig)
+      }
     })
-  })
-  const pressValue = useDerivedValue(() => {
-    return withTiming(isPressing ? 1 : 0, {
-      duration: 120,
-      easing: Easing.bezier(0.44, 0, 0.56, 1)
+    .onFinalize(() => {
+      pressed.value = withTiming(0, animationConfig)
+      if (isSelected) {
+        selected.value = withTiming(0, animationConfig)
+      }
     })
-  })
 
-  const rootStyle = useAnimatedStyle(() => {
-    const backgroundColor = interpolateColor(
-      selectValue.value,
-      [0, 1],
-      [color.background.white, color.secondary.s400]
-    )
+  const longPress = Gesture.LongPress()
+    .minDuration(animationConfig.duration)
+    .onBegin(() => {
+      pressed.value = withTiming(1, animationConfig)
+      if (!isSelected) {
+        selected.value = withTiming(1, animationConfig)
+      }
+    })
+    .onFinalize(() => {
+      pressed.value = withTiming(0, animationConfig)
+      if (isSelected) {
+        selected.value = withTiming(0, animationConfig)
+      }
+    })
 
-    const scale = interpolate(pressValue.value, [0, 1], [1, 0.97])
+  const taps = Gesture.Exclusive(longPress, tap)
 
-    return {
-      backgroundColor,
-      transform: [{ scale }]
-    }
-  })
+  useEffect(() => {
+    setIsSelected(isSelectedProp)
+    selected.value = withTiming(isSelectedProp ? 1 : 0, animationConfig)
+  }, [isSelectedProp, selected])
 
-  const textStyle = useAnimatedStyle(() => {
-    const textColor = interpolateColor(
-      selectValue.value,
+  const animatedRootStyles = useAnimatedStyle(() => ({
+    opacity: withTiming(disabled ? 0.45 : 1, animationConfig),
+    backgroundColor: interpolateColor(
+      selected.value,
+      [0, 1],
+      [color.background.white, color.secondary.s400]
+    ),
+    borderColor: interpolateColor(
+      selected.value,
+      [0, 1],
+      [color.border.strong, color.secondary.s400]
+    ),
+    transform: [{ scale: interpolate(pressed.value, [0, 1], [1, 0.97]) }]
```

Comment:
I don't love how buried these animation values get in here, but I can't think of a great way to model them as variables 😓